src/lsm/lsm_cursor_bulk.c

```diff
@@ -31,7 +34,16 @@ __clsm_close_bulk(WT_CURSOR *cursor)
 	clsm->nchunks = 0;
 
 	/* Set ondisk, and flush the metadata */
-	F_SET(lsm_tree->chunk[0], WT_LSM_CHUNK_ONDISK);
+	F_SET(chunk, WT_LSM_CHUNK_ONDISK);
+	/*
+	 * Setup a generation in our chunk, so that future LSM merges choose
+	 * reasonable sets of chunks.
+	 */
+	avg_chunks = (lsm_tree->merge_min + lsm_tree->merge_max) / 2;
+	for (last_size = avg_chunks * lsm_tree->chunk_size;
+	    last_size < chunk->size;
+	    last_size *= avg_chunks, ++chunk->generation) {}
+
```

Comment:
I find @michaelcahill's suggestion more readable.  I will make that change, test and merge.